include/pybind11/pytypes.h

```diff
@@ -1167,6 +1178,20 @@ class bytes : public object {
             pybind11_fail("Unable to extract bytes contents!");
         return std::string(buffer, (size_t) length);
     }
+
+#ifdef PYBIND11_HAS_STRING_VIEW
+    // NOLINTNEXTLINE(google-explicit-constructor)
+    bytes(std::string_view s) : bytes(s.data(), s.size()) { }
+
+    operator std::string_view() const {
```

Comment:
This is a comment from an unfinished Google-internal PR Ken Oslund (sorry I cannot find his github handle) sent me a long time ago:
```
    // Python retains ownership of the string, so returned view is only valid
    // as long as the python object lives.
```

I think some version of this would be very useful to add to the added `operator std::string_view` in this PR.

Reply:
Added.